manager/volume.go

```diff
@@ -622,6 +625,38 @@ func (m *VolumeManager) Expand(volumeName string, size int64) (v *longhorn.Volum
 		return v, nil
 	}
 
+	pvcName := v.Status.KubernetesStatus.PVCName
+	resourceSize := resource.MustParse(strconv.FormatInt(size, 10))
+	if pvcName != "" {
+		pvcNamespace := v.Status.KubernetesStatus.Namespace
+		pvc, err := m.ds.GetPersisentVolumeClaim(pvcNamespace, pvcName)
+		if err != nil {
+			return nil, err
+		}
+
+		// Should change to pvc.Status.Capacity.Storage(), once upgrade API to v0.18.x
+		if pvc.Status.Capacity.StorageEphemeral().Cmp(resourceSize) > 0 {
+			return nil, fmt.Errorf("cannot expand volume %v with current size %v to a smaller size %v", v.Name, v.Spec.Size, size)
+		}
+
+		// Should change to pvc.Status.Capacity.Storage(), once upgrade API to v0.18.x
+		if pvc.Status.Capacity.StorageEphemeral().Cmp(resourceSize) == 0 {
```

Comment:
Need to check both spec and status

Reply:
Currently, using v0.16.15, only support pvc.Spec.XXX_Size()

Reply:
We are currently using api v16.15 as in https://github.com/kubernetes/api/blob/release-1.16/core/v1/resource.go, PVC spec/status didn't support `Storage` yet.  We already check disk size in earlier code.